Cue/app/api/Library.js

```diff
@@ -69,16 +72,23 @@ var LibraryApi = {
     return CueApi.fetch(endpoint, 'GET')
   },
 
-	addDeckToLibrary(uuid: string) {
-		let endpoint = '/api/v1/library/add';
-		let method = 'POST'
-		let body = JSON.stringify({
-				uuid: uuid,
-				device: DeviceInfo.getDeviceName(),
-			});
-		console.info('Adding deck "' + uuid + '" to library from endpoint: ' + endpoint)
-		return CueApi.fetch(endpoint, method, body);
-	},
+  getUuidByShareCode(shareCode: string) {
+    let endpoint = '/api/v1/deck/uuid/' + shareCode
+    console.info('Fetching deck UUID from share code "' + shareCode + '" at endpoint:' + endpoint)
+    return CueApi.fetch(endpoint)
+  },
+
+  addDeckToLibrary(uuid: string, shareCode?: string) {
+    let endpoint = '/api/v1/library/add';
+    let method = 'POST'
+    let body = JSON.stringify({
+      uuid: uuid,
+      device: DeviceInfo.getDeviceName(),
+      share_code: shareCode || null,
```

Comment:
Nah, if you have `undefined` in an object, then the key just doesn't appear at all in the JSON string.